src/entries/popup/handlers/background.ts

```diff
@@ -1,21 +1,94 @@
 import { createORPCClient } from '@orpc/client';
 import { RPCLink } from '@orpc/client/message-port';
+import {
+  ClientRetryPlugin,
+  ClientRetryPluginContext,
+} from '@orpc/client/plugins';
 import type { RouterClient } from '@orpc/server';
 import { createTanstackQueryUtils } from '@orpc/tanstack-query';
+import * as Sentry from '@sentry/react';
 
 import type { PopupRouter } from '~/entries/background/procedures/popup';
 import { POPUP_PORT_NAME } from '~/entries/background/procedures/popup/constants';
 
 import { createDeepProxy } from './deepProxy';
 import { autoReconnect } from './retry';
 
+interface ORPCClientContext extends ClientRetryPluginContext {}
+
 function createPort() {
   return chrome.runtime.connect({ name: POPUP_PORT_NAME });
 }
-function createClient(port: chrome.runtime.Port): RouterClient<PopupRouter> {
+function createClient(
+  port: chrome.runtime.Port,
+): RouterClient<PopupRouter, ORPCClientContext> {
+  let isClosed = false;
+  port.onDisconnect.addListener(() => {
+    isClosed = true;
+  });
+
   return createORPCClient(
     new RPCLink({
       port,
+      plugins: [
+        new ClientRetryPlugin({
+          default: {
+            // retry 3 times (after 150ms, 300ms and 450ms; max total 900ms)
+            retry: 3,
+            retryDelay(t) {
+              return (t.attemptIndex + 1) * 150;
+            },
+            // only retry if the error is a Not Found error, which indicates the background process is not yet ready
+            shouldRetry({ error }) {
+              return Boolean(
+                error instanceof Error && error.message === 'Not Found',
```

Comment:
Would these be a specific type; doesn't matter either way. Just wondering if we can be more confident vs another generic Not Found error bubbling up from a different code path